Enterprise messaging update (#5)
payara/Payara-Enterprise-Documentation: docs/modules/ROOT/pages/general-info/general-info.adoc

[[overview]]
= Documentation Overview

This page provides some general information about the Payara Platform Enterprise Edition.

[[release-strategy]]
== Release Strategy

> The Payara Platform Community Edition project is released each month. Each release incorporates fixes and enhancements from the Payara team, GlassFish upstream, and Enterprise only features.

Reviewer: Should be Enterprise edition instead

Reviewer: Also: "Enterprise-only"

Reviewer: Also, the Glassfish upstream cherry pick section was removed, so it might be better to remove mention of it here too.

[[new-naming-strategy]]
== New Naming Strategy (June 2020)

As of June 2020, The naming strategy of the Payara Platform Enterprise Edition releases is changed.

The general format of the version number is **Major Version Number** . **Minor Number** . **Patch Number**

> **Major Version** numbers will start from 5 (the current major version of Payara Platform) and only increment by 1 when a new major version (E.g Payara 6) is released.  Fundamental changes (e.g. Javax to Jakarta breaking change) will only be released in Major Versions. There is currently no set cadence for this version change. The change of a major version number is a product management and marketing decision.

Reviewer 2: ```suggestion
**Major Version** numbers will start from 5 (the current major version of Payara Platform) and only increment by 1 when a new major version (E.g Payara 6) is released.  Fundamental changes (e.g. Javax to Jakarta breaking change) will only be released in Major Versions. There is no set cadence for the major version change. 
```
I think simpler is better. No need to mention who decides on major version change.

Author: It was documented as such on the confluence pages.

> Customers should expect changes to their applications and configuration scripts to move to a new Major version.

Reviewer 2: ```suggestion
Customers should expect changes to their applications and configuration scripts when upgrading to a new Major version.
```

> **Minor Version** numbers will start from 20 (the year that Payara Enterprise and Community versions diverge)

Reviewer 2: ```suggestion
**Minor Version** numbers will start from 20
```
No need to mention the reason for starting with 20 in the docs, it adds no value to customers.

Author: Is documented as such on Confluence as messaging.

> and increment by 1 quarterly.  These version changes will introduce significant features (typically originating from the Product Management process or stabilised and adopted from Community releases). 

Reviewer 2: ```suggestion
and increment by 1 quarterly.  These version changes will introduce new features and enhancements 
```
No need to mention how we choose the features to add, no value for the customer.

Author: Is documented as such on Confluence as messaging.

Features introduced during a minor version change must be backwards compatible with previous minor releases with no intervention required by an operator. 

> Features introduced during a minor version change can only introduce breaking changes as a last resort and with product management approval and appropriate customer messaging.

Reviewer 2: ```suggestion
Features introduced during a minor version change can only introduce breaking changes as a last resort and with a special caution. In that case, customers will be sufficiently informed about it and guided how to upgrade with minimal impact on their systems.
```

Again, no need to mention product management. The term "customer messaging" seems a bit alien to customers so I suggest a reworded sentence.

The Minor Version number will reset to 0 on a Major Version release.

Customers should not expect to change their applications and configuration scripts to move to a minor version unless by choice to take advantage of newly introduced features.

> **Patch Version** numbers will start from 0 and as a minimum increment monthly.  These version change will introduce bug fixes and patches.  The Patch Version number will reset to 0 on a Minor Version release.

Reviewer 2: ```suggestion
**Patch Version** numbers will start from 0 and increment monthly after each release. This version change will introduce bug fixes and patches. The Patch Version number will reset to 0 on a Minor Version release.
```

Customer specific versions will add an additional 4th level number where the number can reflect something meaningful to the customer e.g. the number of the hotfix. For example  5.20.5.1234

Customers should not expect to change their applications and configuration scripts to move to a patch version.

[cols=-2*,options="header"]
|=== 

| Release date| Version  

| June 2020 | 5.20.0

| July 2020 | 5.20.1

| August 2020 | 5.20.2

| September 2020 | 5.21.0

| ... | 5.21.1

| ...| ...

| March 2021 | 5.23.0

| April 2021 | 5.23.1

| May 2021 | 5.23.2

| June 2021 | 5.24.0

| ... | 5.24.1

| ...| ...

|=== 

> [[naming-strategy]]

Reviewer 2: This is a naming strategy for Payara 4, is it relevant? I would describe the previous naming strategy for Payara 5, e.g. 5.201, 5.194, ...

Author: This is already on the documentation page today in this format. We can change this with another PR unrelated to split Community / Enterprise.
